### PR TITLE
[MNG-6271] Add validationRule for expressions in repositoryUrl

### PR DESCRIPTION
**Pull request by rfscholte**

### CODE REVIEW

File: core-it-suite/src/test/resources/mng-6759-transitive-dependency-repositories/module2/pom.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <repositories>
         <repository>
             <id>customRepo</id>
-            <url>${custom.repo.uri}</url>
+            <url>file:///${basedir}/../dependency-in-custom-repo/target/repo</url>
```
Comment (michael-o) [inline core-it-suite/src/test/resources/mng-6759-transitive-dependency-repositories/module2/pom.xml:14]: Isn't this deprecated for `{project.basedir}`?

Comment (rfscholte) [reply core-it-suite/src/test/resources/mng-6759-transitive-dependency-repositories/module2/pom.xml:14]: According to https://maven.apache.org/ref/current/maven-core/apidocs/org/apache/maven/plugin/PluginParameterExpressionEvaluator.html it is still valid

Comment (michael-o) [reply core-it-suite/src/test/resources/mng-6759-transitive-dependency-repositories/module2/pom.xml:14]: My bad, it is indeed still valid...

**Review by michael-o — CHANGES_REQUESTED** [review]
Plaese also change the commit message here..

**Comment by rfscholte** [comment]
Will adjust the commit message after the squash and merge.

**Pull request closed without merging**